fools/py/api_client.py

```
@@ -0,0 +1,193 @@
+
+        data = None
+        if body is not None:
+            data = json.dumps(body).encode("utf-8")
+
+        request = Request(url, data=data, headers=merged_headers, method=method)
+
+        last_error = None
+        for attempt in range(self.retry_config.max_retries + 1):
+            start = time.monotonic()
+            try:
+                with urlopen(request) as resp:
```

Ruff (0.15.13)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)